### PR TITLE
FE: Monitoring Policies Edit Panel - initial 

### PR DESCRIPTION
**Pull request by faizanjanjua**
## Description
Create a components/MonitoringPoliciesEditPanel component, route will be /monitoring-policies-new/:id where id is a number, the database id of the policy. Note that this route must be after /monitoring-policies-new/create.

When a user is on the MonitoringPoliciesDetailPanel and clicks the Edit icon, they will route to this component.

For this ticket:

update route to use this component

ensure component is navigated-to from main table, but only after LOK-2380: FE: Monitoring Policies - Details panel, initial version - Bilal
DONE
is done. monitoringPolicies.selectedPolicy should be set

display status (cannot actually change it)

add Basic Information for name and description (memo field)

add section names (placeholders) for Tags, Notifications, Alert Rules, but don’t implement them

other sections will be added in separate tickets

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2420

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/components/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const store = useMonitoringPoliciesStore()
+const tagQueries = useTagQueries()
+
+</script>
+<style lang="scss" scoped>
+@use '@featherds/styles/themes/variables';
+@use '@featherds/styles/mixins/typography';
+@use '@/styles/vars.scss';
+
+
```
Comment (synqotik) [inline ui/src/components/MonitoringPolicies/MonitoringPolicyBasicInformationEditAddForm.vue:34]: Remove 2 blank lines, so just one blank line between `@use` and first class

File: ui/components.d.ts [hunk 2]
```diff
@@ -115,6 +115,8 @@ declare module '@vue/runtime-core' {
     MonitoringPoliciesCardLegacy: typeof import('./src/components/MonitoringPoliciesLegacy/MonitoringPoliciesCardLegacy.vue')['default']
     MonitoringPoliciesCardThresholdConditionTableLegacy: typeof import('./src/components/MonitoringPoliciesLegacy/MonitoringPoliciesCardThresholdConditionTableLegacy.vue')['default']
     MonitoringPoliciesDetailPanel: typeof import('./src/components/MonitoringPolicies/MonitoringPoliciesDetailPanel.vue')['default']
+
```
Comment (synqotik) [inline ui/components.d.ts:118]: Remove blank line

**Review by synqotik — CHANGES_REQUESTED** [review]
Overall looks good, mostly formatting changes and a few small code changes.

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesDetailPanel.vue [hunk 3]
```diff
@@ -124,8 +126,12 @@ const deleteMsg = computed(() =>
   `Deleting monitoring policy ${store.selectedPolicy?.name} removes ${store.numOfAlertsForRule} associated alerts. Do you wish to proceed?`
 )
 
-const onEdit = () => {
+const onEdit = (id: number) => {
   console.log('onEdit clicked')
+  if (id != undefined) {
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesDetailPanel.vue]: `
const onEdit = (id: number) => {  
  if (id) {
    router.push(`/monitoring-policies-new/${id}`)
  }
}`

changed as per your instructions

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue [hunk 4]
```diff
@@ -1,15 +1,103 @@
...
+import { useMonitoringPoliciesStore } from '@/store/Views/monitoringPoliciesStore'
+import { TagSelectItem } from '@/types'
+import { Policy } from '@/types/policies'
+import BasicAutocomplete from '../Common/BasicAutocomplete.vue'
+import router from '@/router'
+
+const handleCancel = () => {
+  router.push('/monitoring-policies-new/')
+}
+const store = useMonitoringPoliciesStore()
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue]: chagned order to   imports, then functions, then watchEffect

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue [hunk 5]
```diff
@@ -1,15 +1,103 @@
...
+const populateForm = (policy: Policy) => store.displayPolicyForm(policy)
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const formattedTags = computed(() => store.selectedPolicy!.tags!.map((tag: string) => ({ name: tag, id: tag })))
+const route = useRoute()
+
+watchEffect(() => {
+  if (store.monitoringPolicies.length > 0 && route?.params?.id) {
+    const filteredPolicy = store.monitoringPolicies.find((item: Policy) => item.id === Number(route.params.id))
+    populateForm(filteredPolicy as Policy)
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue]: watchEffect(() => {
  if (store.monitoringPolicies.length > 0 && route?.params?.id) {
    const filteredPolicy = store.monitoringPolicies.find((item: Policy) => item.id === Number(route.params.id))
    store.displayPolicyForm(filteredPolicy as Policy)
    }
})

File: ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue [hunk 6]
```diff
@@ -1,15 +1,103 @@
...
+      </template>
+      <FeatherTabPanel class="bg-white mt-0 p-2">
+        <!-- copy Initial from Here -->
+        <MonitoringPolicyBasicInformationEditAddForm />
+        <div class="d-flex justify-content-end mt-2">
+          <FeatherButton text @click="handleCancel">Cancel</FeatherButton>
+          <FeatherButton primary>Next</FeatherButton>
+        </div>
+      </FeatherTabPanel>
+      <FeatherTabPanel>TWOTOOW</FeatherTabPanel>
```
[the remark this answers is not in the document]
Comment (faizanjanjua) [reply ui/src/components/MonitoringPolicies/MonitoringPoliciesEditPanel.vue]: Done

**Review by synqotik — APPROVED** [review]
LGTM

**Pull request merged**